server.js

```diff
@@ -16,11 +19,107 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
+const ThoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140,
+    trim: true
+  },
+  createdAt: {
+    type: Date,
+    default: () => new Date()
+  },
+  hearts: {
+     type: Number,
+     default: 0
+  }
+
+})
+
+const Thought = mongoose.model('Thought', ThoughtSchema)
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
 });
 
+// Get all messages/thoughts
+
+app.get('/thoughts', async (req,res) => {
+
+try {
+   const thoughts = await Thought.find()
+   .sort({ 
+    createdAt: 'desc'
+  })
+  .limit(20)
+  .exec()
+
+  res.status(200).json(thoughts)
+
+} catch (error) {
+  res.status(400).json({success: false, response: error})
+}
+})
+
+//Post a new thought
+
+app.post('/thoughts', async (req, res) => {
+
+const { message } = req.body
+
+const thought = await new Thought({ message: message}).save()
```

Comment:
Just realized that I declared and saved the thought variable inside the try() function, I wonder what difference it makes in the outcome!